cal/cal/helpers.py

```diff
@@ -113,9 +119,34 @@ def get_time_series(model, timezone='UTC', time_step='week', calendar_ids=None,
             else:
                 total += (events[i].end - events[i].start).total_seconds() / 3600
             i += 1
+
+        # For moving average. Basically iterates back and adds average
+        data_point += 1
+        period_total += total
+        if data_point % period == 0:
+            average = float(period_total) / period
+            period_start = start
+            sub_range = []
+            for _ in range(period):
+                sub_range = [(period_start, average)] + sub_range
+                period_start -= increment
+            moving_average_lst += sub_range
+            period_total = 0
+
         week_hours.append((start, total))
         start = end
-    return week_hours
+
+    # Deals with edge case if number of data points not divisible by period
+    if data_point % period != 0:
+        average = float(period_total) / (data_point % period)
+        period_start = start - increment
+        sub_range = []
+        for _ in range(data_point % period):
+            sub_range = [(period_start, average)] + sub_range
+            period_start -= increment
+        moving_average_lst += sub_range
+
+    return (week_hours, moving_average_lst)
 
```

Comment:
hmm I think with logic-heavy algorithms like this, you should add documentation of what your logic so that if anyone in the future needs to change/fix something, they would know what u did